apps/meteor/app/lib/server/functions/saveUser.ts

```diff
@@ -40,50 +47,48 @@ Meteor.startup(() => {
 	});
 });
 
-async function _sendUserEmail(subject, html, userData) {
+async function _sendUserEmail(subject: string, html: string, userData: RequiredField<ISaveUserDataParams, 'email'>) {
 	const email = {
 		to: userData.email,
-		from: settings.get('From_Email'),
+		from: settings.get<string>('From_Email'),
 		subject,
 		html,
 		data: {
 			email: userData.email,
 			password: userData.password,
+			...(userData.name && { name: userData.name }),
 		},
 	};
 
-	if (typeof userData.name !== 'undefined') {
-		email.data.name = userData.name;
-	}
-
 	try {
 		await Mailer.send(email);
-	} catch (error) {
+	} catch (error: any) {
 		throw new Meteor.Error('error-email-send-failed', `Error trying to send email: ${error.message}`, {
 			function: 'RocketChat.saveUser',
 			message: error.message,
 		});
 	}
 }
 
-async function validateUserData(userId, userData) {
+async function validateUserData(userId: IUser['_id'], userData: ISaveUserDataParams) {
 	const existingRoles = _.pluck(await getRoles(), '_id');
+	const isUpdateUserParams = '_id' in userData && userData._id;
```

Comment:
hmmm this does make sense, but we can't do that because it'd cause a typecheck error
`ISaveUserDataParams` may be `UserCreateParamsPOST`, which doesn't have the `_id` prop. So if we try to "access" it directly  (as you suggested) without checking if `'_id' in userData`, then we'd have an error (`the _id property does not exist in the UserCreateParamsPOST type`, even if we try `!!userData?._id`)

Reply:
If `userData` can be of 2 types (union) I'd recommend doing this check into a type check func
```
isUserCreateParams = (d): d is UserCreateParams => ...

Reply:
good idea! done, looks much better now (I had to keep checking for `'_id' in userData` due to the restriction I mentioned, but this still looks better)